### PR TITLE
ENH: Improve plan md.

### PR DESCRIPTION
**Pull request by danielballan**
- Use ChainMap to give md from kwargs precedence over hard-coded md.
- Add 'plan_name' to everything.
- Add a couple 'plan_args' accidentally left out in the first pass.

### CODE REVIEW

File: bluesky/plans.py [hunk 1]
```diff
@@ -1790,6 +1809,9 @@ def relative_log_scan(detectors, motor, start, stop, num, *, per_step=None,
     `bluesky.plans.log_scan`
     """
     # TODO read initial positions (redundantly) so they can be put in md here
+    if md is None:
+        md = {}
+    md = ChainMap(md, {'plan_name': 'relative_log_scan'})
```
Comment (tacaswell) [inline bluesky/plans.py]: This is not going to propagate correctly

File: bluesky/plans.py [hunk 2]
```diff
@@ -1833,6 +1855,22 @@ def adaptive_scan(detectors, target_field, motor, start, stop,
...
+         'motors': [motor.name],
+         'plan_args':{'detectors': list(map(repr, detectors)),
+                      'motor': repr(motor),
+                      'start': start,
+                      'stop': stop,
+                      'min_step': min_step,
+                      'max_step': max_step,
+                      'target_delta': target_delta,
+                      'backstep': backstep,
+                      'threshhold': threshold},
```
Comment (klauer) [inline bluesky/plans.py]: typo

File: bluesky/plans.py [hunk 3]
```diff
@@ -1985,10 +2026,14 @@ def scan_nd(detectors, cycler, *, per_step=None, md=None):
     """
     if md is None:
         md = {}
-    md.update({'detectors': [det.name for det in detectors],
-               'motors': [motor.name for motor in cycler.keys]})
-    md['plan_args'] = {'detectors': list(map(repr, detectors)),
-                       'cycler': repr(cycler)}
+    md = ChainMap(
+        md,
+        {'detectors': [det.name for det in detectors],
```
Comment (klauer) [inline bluesky/plans.py]: Elsewhere `repr()` is used on detectors, why `name` here? (or why `repr` elsewhere...)

File: bluesky/plans.py [hunk 4]
```diff
@@ -1624,6 +1630,9 @@ def relative_list_scan(detectors, motor, steps, *, per_step=None, md=None):
     `bluesky.plans.list_scan`
     """
     # TODO read initial positions (redundantly) so they can be put in md here
+    if md is None:
+        md = {}
+    md = ChaimMap(md, {'plan_name': 'relative_list_scan'})
```
Comment (klauer) [inline bluesky/plans.py]: Chaim

File: bluesky/plans.py (hunk 3 above)
Comment (klauer) [inline bluesky/plans.py]: I blame it on lack of sleep. I get it now: top-level metadata gets the name, plan_args gets the repr

File: bluesky/plans.py (hunk 1 above)
Comment (danielballan) [inline bluesky/plans.py]: It is; in an offline-conversion, @tacaswell established that the precedence order is the opposite of what he was thinking here.

**Pull request merged**